### PR TITLE
Strict clx defclass

### PR DESCRIPTION
**Pull request by dkochmanski**

### CODE REVIEW

**Comment by JMC-design** [comment]
I would think the error from ecase above conveys all the necessary information, no?
I probably should have done something about the formatting of that file before. Just remember I had to piece it together from fragments recovered from my dead harddrive, then my laptop died and I lost my housing, so the pieces were stitched together on the web. Strangely enough the same thing happened last summer, so I've gotten a bit apprehensive of working on X since my harddrives always crash and die and I become homeless.

**Comment by dkochmanski** [comment]
change is in the "else" branch of if. do you suggest that instead of assert we should put:

`(dolist (option options) (ecase (car option) (ecase ((:a :b :c) #| empty body |#)))`?

**Comment by JMC-design** [comment]
lol, oh right, and there's one of the reasons why I don't like functions/macros that can't fit on my screen all at once. ...and monday mornings.

File: depdefs.lisp [hunk 1]
```diff
@@ -542,9 +542,16 @@ used, since NIL is the empty list.")
...
 		      ((object ,name) stream)
 		      (,print-function object stream 0))))))))
-      `(within-definition (,name def-clx-class)
-	 (defstruct (,name ,@options)
-	   ,@slots))))
+      (flet ((assert-valid-option (option)
+               (unless (and (listp option)
+                            (member (first option)
+                                    '(:constructor :include :print-function
+                                      :copier :predicate)))
```
Comment (scymtym) [inline depdefs.lisp]: Could use `(unless (typep option '(cons (member :constructor …))) …)`.

Comment (dkochmanski) [reply depdefs.lisp]: sure, done.

**Pull request merged**